src/libs/relay/conduit_relay_io_silo.cpp

```diff
@@ -4003,6 +4154,103 @@ write_multimats(DBfile *dbfile,
     }
 }
 
+//-----------------------------------------------------------------------------
+// only for overlink
+void
+write_var_attributes(DBfile *dbfile, 
+                     const std::string &opts_mesh_name,
+                     const Node &root)
+
+{
+    const Node &n_mesh = root["blueprint_index"][opts_mesh_name];
+    const Node &n_type_dom_info = root["type_domain_info"];
+    if (n_mesh.has_child("fields"))
+    {
+        std::vector<std::string> multivar_name_strings;
+        std::vector<int> elemlengths;
+        int nvalues = 0;
+        std::vector<int> var_attr_values;
+
+        auto field_itr = n_mesh["fields"].children();
+        while (field_itr.has_next())
+        {
+            const Node &n_var = field_itr.next();
+            std::string var_name = field_itr.name();
+            std::string safe_varname = detail::sanitize_silo_varname(var_name);
+            multivar_name_strings.push_back(safe_varname);
+
+            const int num_attr = 5; // we are writing 5 var attributes for now
+
+            elemlengths.push_back(num_attr);
+            nvalues += num_attr;
+
+            // 
+            // centering: ATTR_NODAL 0, ATTR_ZONAL 1, ATTR_FACE, ATTR_EDGE
+            // 
+            if (n_var["association"].as_string() == "vertex")
+            {
+                var_attr_values.push_back(0); // nodal == vertex
```

Comment:
Would it help to use the Silo enum types here, instead of the raw values?

Reply:
We don't have access to the raw enum types, as these are Overlink enum values. Even if Overlink and Silo agree on the enum values and meanings (not sure if they do for centering or not), I think it is cleaner to use raw values and clearly label what the Overlink values mean.